test-requirements.txt

```diff
@@ -33,26 +24,20 @@ click==8.1.7
     #   black
     #   pip-tools
 codespell==2.2.6
-    # via -r test-requirements.in
 coverage==7.4.1
-    # via -r test-requirements.in
 cryptography==42.0.2
     # via
-    #   -r test-requirements.in
     #   pyopenssl
     #   trustme
     #   types-pyopenssl
 dill==0.3.8
     # via pylint
 docutils==0.20.1
     # via sphinx
-exceptiongroup==1.2.0 ; python_version < "3.11"
-    # via
-    #   -r test-requirements.in
-    #   pytest
+exceptiongroup==1.2.0
+    # via pytest
```

Comment:
This change too seems wrong

Reply:
I think it's intended, as we specify `--python-version=3.8`. If we generate separate `test-requirements-pyXX.txt` files for each python version we test we'll only get `exceptiongroup` in the ones that are <3.11

Reply:
So I think that means we should have a `requirements` directory wherein we generate txt files for all versions we test.